src/nickel.rs

```diff
@@ -273,45 +247,64 @@ mod ssl {
     /// extern crate hyper;
     ///
     /// use nickel::Nickel;
+    /// # #[cfg(feature = "ssl")]
     /// use hyper::net::Openssl;
     ///
+    /// # #[cfg(feature = "ssl")]
     /// # fn main() {
     /// let server = Nickel::new();
     /// let ssl = Openssl::with_cert_and_key("foo.crt", "key.pem").unwrap();
     /// server.listen_https("127.0.0.1:6767", ssl);
     /// # }
+    /// # #[cfg(not(feature = "ssl"))]
+    /// # fn main() {}
```

Comment:
Originally TLS support in Hyper was hardcoded to rust-openssl, but that was changed so that Hyper could deal with any TLS backend through a couple of traits. For historical reasons, Hyper's `ssl` feature is the one that enables the implementations of those traits for rust-openssl, but it should really be renamed at some point.

Because of this, I think the the nickel feature can be removed entirely now, actually - Nickel doesn't need to care about what `SslServer` implementations exist, other than that *some* implementation is being given to `Nickel::listen_https`.

That does complicate this example a bit, though - it should maybe be just tagged `ignore` at that point so it doesn't need to compile.

As far as what ssl implementations people should use, rust-openssl's probably the right thing to recommend for unixy platforms other than OSX. OSX ships with a copy of OpenSSL, but it's *super* old and Apple is fairly aggressively trying to kill it off. In particular, El Capitan no longer includes OpenSSL headers, which prevents rust-openssl from compiling so you have to pull a copy in with something like Homebrew.

The bindings for OSX's native Secure Transport APIs are in a pretty decent place I think, I'd probably recommend that on OSX for most simple use cases.

On Windows, OpenSSL is even more of a nightmare to get set up, but the bindings to SChannel, Windows's native SSL API, still need a lot of work, so you're pretty much stuck with OpenSSL there, unfortunately.